### PR TITLE
Rewrite ProjectMuse using EM

### PR DESCRIPTION
**Pull request by adam3smith**
No more RIS available. Books metadata is limited -- no place to be found anywhere.
For PDF attachments, I do overwrite all attachments on purpose; I also leave the PDF URL in on purpose.
@zuphilip whenever you can take a look?

### CODE REVIEW

File: Project MUSE.js [hunk 1]
```diff
@@ -1,182 +1,116 @@
...
+    // Embedded Metadata
+    translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+    translator.setDocument(doc);
+    translator.setHandler('itemDone', function (obj, item) {
+        if (abstract) {
+        	item.abstractNote = abstract.replace(/^\s*Abstract/, "").replace(/,\s*$/, "").trim();
+        }
+        if (url.indexOf("/article/") != -1) {
+       		var pdfurl = url.replace(/(\/article\/\d+).*/, "$1") + "/pdf";
+       		Z.debug(pdfurl)
```
Comment (zuphilip) [inline Project MUSE.js]: Delete or comment out

File: Project MUSE.js [hunk 2]
```diff
@@ -1,182 +1,116 @@
...
-		var m = hostRe.exec(url);
-		var host = m[1];
+function scrape(doc, url) {
+	var abstract = ZU.xpathText(doc, '//div[@class="abstract"]');
+	if (!abstract) abstract = ZU.xpathText(doc, '//div[@class="description"][1]/text()');
+	var translator = Zotero.loadTranslator('web');
+    // Embedded Metadata
+    translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+    translator.setDocument(doc);
+    translator.setHandler('itemDone', function (obj, item) {
```
Comment (zuphilip) [inline Project MUSE.js]: The indent seems on these 4 lines wrong. Maybe check also above that you are using tabs and not spaces for the indent (when I copy and paste the code block this goes usually wrong).

File: Project MUSE.js [hunk 3]
```diff
@@ -1,182 +1,116 @@
...
+        	item.abstractNote = abstract.replace(/^\s*Abstract/, "").replace(/,\s*$/, "").trim();
+        }
+        if (url.indexOf("/article/") != -1) {
+       		var pdfurl = url.replace(/(\/article\/\d+).*/, "$1") + "/pdf";
+       		Z.debug(pdfurl)
+       		item.attachments = [{
+						"url": pdfurl,
+						"title": "Full Text PDF",
+						"mimeType": "application/pdf"
+					}]
```
Comment (zuphilip) [inline Project MUSE.js]: Semicolon is missing here.

BTW I prefer to use here also `item.attachments.push( { ... } );`, which will not destroy any other attachments possibly added at some time.

File: Project MUSE.js [hunk 4]
```diff
@@ -1,182 +1,116 @@
...
-			item.complete();
-			});
-			translator.translate();
-			})
-	}
-	
-	else{
+    translator.getTranslatorObject(function(trans) {
+        trans.doWeb(doc, url);
+    });
```
Comment (zuphilip) [inline Project MUSE.js]: If you don't need any special handling here, then you can omit this function completely.

File: Project MUSE.js [hunk 5]
```diff
@@ -190,62 +124,53 @@ var testCases = [
...
+				"abstractNote": ", , Jeffrey S. Theis focuses on pastoral literature in early modern England as an emerging form of nature writing. In particular, Theis analyzes what happens when pastoral writing is set in forests — what he terms “sylvan pastoral.”",
```
Comment (zuphilip) [inline Project MUSE.js]: Why does this start with two commas? Are there some empty nodes?

File: Project MUSE.js [hunk 6]
```diff
@@ -190,62 +124,53 @@ var testCases = [
...
-	},
-	{
-		"type": "web",
-		"url": "http://muse.jhu.edu/books/9780820705057",
+		"url": "http://muse.jhu.edu/book/785",
 		"items": [
 			{
 				"itemType": "book",
-				"title": "Writing the Forest in Early Modern England: A Sylvan Pastoral Nation",
+				"title": "Writing the Forest in Early Modern England",
```
Comment (zuphilip) [inline Project MUSE.js]: It is a pity, that we lose the subtitle here...

**Comment by zuphilip** [comment]
This looks great. I have just some small comments, see above.

File: Project MUSE.js (hunk 4 above)
Comment (zuphilip) [inline Project MUSE.js]: Or do you do this because of https://github.com/zotero/zotero/issues/859 ?

Comment (adam3smith) [inline Project MUSE.js]: yes, but I didn't text if it's necessary.

Comment (zuphilip) [inline Project MUSE.js]: `s/text/test/` ? :smirk:

**Comment by adam3smith** [comment]
if you want to take another look @zuphilip ?

**Comment by zuphilip** [comment]
This looks fine for me!

(BTW for calling web translator in translation server the function `setDocument` is currently broken, but `getTranslatorObject` seems a way to circumvent this...)

**Comment by adam3smith** [comment]
> (BTW for calling web translator in translation server the function setDocument is currently broken, but getTranslatorObject seems a way to circumvent this...)

Do I understand that correctly that if I replace `translator.translate()` with 

```
translator.getTranslatorObject(function(trans) {
        trans.doWeb(doc, url);
   });
```

That this will work on the server, but otherwise it won't? In that case, I should probably switch this back, no?

**Comment by zuphilip** [comment]
Yes, but you need additionally also to delete the `translator.setDocument(doc);` line (that is the critical function), cf. https://github.com/zotero/translation-server/issues/26#issuecomment-241850317

**Pull request merged**